security/pkg/pki/util/keycertbundle.go

```diff
@@ -318,18 +319,12 @@ func Verify(certBytes, privKeyBytes, certChainBytes, rootCertBytes []byte) error
 	return nil
 }
 
-func extractCertExpiryTimestamp(certType string, certPem []byte) (float64, error) {
+func extractCertExpiryTimestamp(certType string, certPem []byte) (*time.Time, error) {
 	cert, err := ParsePemEncodedCertificate(certPem)
 	if err != nil {
-		return -1, fmt.Errorf("failed to parse the %s: %v", certType, err)
-	}
-
-	end := cert.NotAfter
-	expiryTimestamp := float64(end.Unix())
-	if end.Before(time.Now()) {
-		return expiryTimestamp, fmt.Errorf("expired %s found, x509.NotAfter %v, please transit your %s", certType, end, certType)
+		return nil, fmt.Errorf("failed to parse the %s: %v", certType, err)
```

Comment:
Unsure about this change.
I feel like this function should leave expiration checking to the caller, or mention expiration checking to the caller.
Regardless, I don't think using -1 as a sentinel value for an error value makes much sense since -1 indicates 1 second before new year 1970.